templates/onchain/flake.nix

```diff
@@ -0,0 +1,58 @@
+{
+  description = "A liqwid-nix Plutarch project";
+
+  nixConfig = {
+    extra-experimental-features = [ "nix-command" "flakes" "ca-derivations" ];
+    extra-substituters = [ "https://cache.iog.io" "https://public-plutonomicon.cachix.org" "https://mlabs.cachix.org" ];
+    extra-trusted-public-keys = [ "hydra.iohk.io:f/Ea+s+dFdN+3Y/G+FDgSq+a5NEWhJGzdjvKNGv0/EQ=" "public-plutonomicon.cachix.org-1:3AKJMhCLn32gri1drGuaZmFrmnue+KkKrhhubQk/CWc=" ];
+    allow-import-from-derivation = "true";
+    bash-prompt = "\\[\\e[0m\\][\\[\\e[0;2m\\]liqwid-nix \\e[0;5m\\]2.0 \\[\\e[0;93m\\]\\w\\[\\e[0m\\]]\\[\\e[0m\\]$ \\[\\e[0m\\]";
+    max-jobs = "auto";
+    auto-optimise-store = "true";
+  };
+
+  inputs = {
+    nixpkgs.follows = "liqwid-nix/nixpkgs";
+    nixpkgs-latest.url = "github:NixOS/nixpkgs";
+
+    liqwid-nix = {
+      url = "github:Liqwid-Labs/liqwid-nix/main";
+      inputs.nixpkgs-latest.follows = "nixpkgs-latest";
+    };
+  };
+
+  outputs = { self, liqwid-nix, flake-parts, ... }:
+    flake-parts.lib.mkFlake { inherit self; } {
+      imports = liqwid-nix.allModules ++ [
+        ({ self, ... }:
+          {
+            perSystem = { config, pkgs', self', inputs, system, ... }:
+              let
+                pkgs = import self.inputs.nixpkgs {
+                  inherit system;
+                };
+              in
+              {
+                onchain.default = {
+                  src = ./.;
+                  ghc.version = "ghc925";
+                  shell = { };
+                  enableBuildChecks = true;
+                  extraHackageDeps = [
+                    "${self.inputs.plutarch-numeric}"
+                    "${self.inputs.plutarch-quickcheck}"
+                    "${self.inputs.plutarch-context-builder}"
+                    "${self.inputs.liqwid-plutarch-extra}"
+                    "${self.inputs.liqwid-script-export}"
+                    "${self.inputs.liqwid-script-export.inputs.ply}/ply-core"
+                    "${self.inputs.liqwid-script-export.inputs.ply}/ply-plutarch"
```

Comment:
I think we'll also need `plutarch-unit` for liqwid

Reply:
Ah, this shouldn't be here at all.